### PR TITLE
Add a CHANGELOG entry about allowing collection downcast in a cast pattern

### PR DESCRIPTION
**Pull request by simanerush**
This adds a new CHANGELOG entry related to PR #60682

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,16 @@ _**Note:** This is in reverse chronological order, so newer entries are added to
 
 ## Swift 5.8
 
+* The Swift compiler now no longer prohibits collection downcast in a cast pattern. Here is the usage of a collection downcast pattern in a `switch` statement:
```
Comment (AnthonyLatsis) [inline CHANGELOG.md:7]: Could we rephrase this to say "now supported" rather than "no longer prohibits"? Also, mentioning the compiler is redundant.

Comment (AnthonyLatsis) [inline CHANGELOG.md:7]: Second sentence: "For example" or similar will suffice.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,16 @@ _**Note:** This is in reverse chronological order, so newer entries are added to
...
+
+```swift
+switch arr {
+   case let ds as [Derived]:
+     ()
+   case is [Derived]:
+     ()
+```
```
Comment (AnthonyLatsis) [inline CHANGELOG.md:15]: Let’s make this self-contained, e.g. you could wrap the statement in a function taking an array of `Any`, and downcast it to an array of integers or a different frequently used type.

Comment (AnthonyLatsis) [inline CHANGELOG.md:15]: I think `...` would make more sense than `()`.

Comment (AnthonyLatsis) [inline CHANGELOG.md:15]: (optional) Cases are usually not indented, although this is a stylistic preference ultimately.

File: CHANGELOG.md (hunk 1 above)
Comment (AnthonyLatsis) [inline CHANGELOG.md:7]: Could we add a link to the associated issue? See older entries for an example.

**Review by AnthonyLatsis — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,21 @@ _**Note:** This is in reverse chronological order, so newer entries are added to
 
 ## Swift 5.8
 
+* [#56139][]:
+
+  Сollection downcasts in cast patterns are now supported. For example,
```
Comment (xedin) [inline CHANGELOG.md:9]: Nit: Should there be `:` instead of `,` after “example”?

**Review by xedin — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (simanerush) [reply CHANGELOG.md:9]: That makes sense, I changed it!

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test
